extension/httpfs/httpfs.cpp

```diff
@@ -506,7 +506,7 @@ static optional_ptr<HTTPMetadataCache> TryGetMetadataCache(FileOpener *opener, H
 	} else {
 		auto lookup = client_context->registered_state.find("http_cache");
 		if (lookup == client_context->registered_state.end()) {
-			auto cache = make_shared<HTTPMetadataCache>(true, false);
+			auto cache = make_shared<HTTPMetadataCache>(true, true);
```

Comment:
I took a look at this, main comment is that now HTTPMetadataCache is always multithreaded-safe, so should the second argument just be dropped or maybe have it default to true with a comment saying "you are on your own!"?

Reply:
fair point! However, I would argue it is not worth changing now, it's pretty clear you can use the cache either local or shared. I don't think we need to add comments warning developers to do shared access to non-threadsafe code?

The error here was simply missing the fact that the cache in the clientcontext would be accessed by multiple threads, this would not be solved by changing the metadata cache class.